dashboard/app/models/pd/application/teacher1819_application.rb

```diff
@@ -94,17 +95,36 @@ def save_partner
       self.regional_partner_id = sanitize_form_data_hash[:regional_partner_id]
     end
 
-    before_save :enroll_user, if: -> {properties_changed?}
+    # override
+    def lock!
+      return if locked?
+      super
+      enroll_user if status == "accepted"
+    end
+
     def enroll_user
       return unless pd_workshop_id
 
-      Pd::Enrollment.find_or_create_by!(
+      enrollment = Pd::Enrollment.where(
         pd_workshop_id: pd_workshop_id,
         email: user.email
-      ) do |enrollment|
-        enrollment.user = user
-        enrollment.school_info = user.school_info
-        enrollment.full_name = user.name
+      ).first_or_initialize
+
+      # If this is a new enrollment, we want to:
+      #   - save it with all required data
+      #   - save a reference to it in properties
+      #   - delete the previous auto-created enrollment if it exists
+      if enrollment.new_record?
+        enrollment.update(
+          user: user,
+          school_info: user.school_info,
+          full_name: user.name
+        )
+        enrollment.save!
+
+        Pd::Enrollment.find_by(id: auto_assigned_enrollment_id).try(:destroy) if auto_assigned_enrollment_id
+
+        self.auto_assigned_enrollment_id = enrollment.id
```

Comment:
I wonder, should we also delete the `auto_assigned_enrollment_id` when changing the status from accepted to something else?

Reply:
That seems reasonable. @mehalshah @ryansloan any thoughts?

Reply:
sgtm